.github/workflows/ci.yml

```diff
@@ -0,0 +1,140 @@
+name: CI Workflow
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - rel/**
+  pull_request:
+    branches:
+      - main
+
+permissions: read-all
+
+jobs:
+  lint:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Cache pre-commit
+        uses: actions/cache@v4
+        with:
+          path: ${{ runner.home }}/.cache/pre-commit
+          key: ${{ runner.os }}-pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}
+          restore-keys: ${{ runner.os }}-pre-commit-
```

Comment:
```suggestion
```

Nit: The runtime without cache is <1min. We could as well just remove the cache step here to keep it simple.